src/js/jsx/sections/layers/LayerFace.jsx

```diff
@@ -147,9 +183,15 @@ define(function (require, exports, module) {
                 documentStore = this.getFlux().store("document"),
                 currentDocument = documentStore.getDocument(documentID),
                 currentLayer = currentDocument.layers.byID(this.props.layer.id);
-
-            this.getFlux().actions.groups.setGroupExpansion(currentDocument, currentLayer,
-                !currentLayer.expanded, descendants);
+            
+            // Performance Hack: delay the expand action to promote browser paint event.
+            Promise
+                .bind(this)
+                .delay(10)
```

Comment:
There was a bug in bluebird 3.0.x such that zero-delay asynchronous callbacks were executed with `setTimeout` instead of something functionally equivalent to `setImmediate`. Even though `setImmediate` is only available in Edge, it's possible to write a function using different primitives (e.g., `MutationObserver`) with the same semantics. This is what bluebird should have been doing all along, but only started doing again in 3.1.x as a result of a bugfix. This very subtle change in behavior has triggered a few bugs in our code already (in cases where we were inappropriately sensitive to the difference).

Anyway!